bingo.js

```diff
@@ -602,9 +675,119 @@ function setSquareColor(square, colorClass)
 {
 	ALL_COLOURS.forEach(c => square.removeClass(c));
 	square.addClass(colorClass);
+	updateCurrentSheetProgress();
 }
 
-function copySeedToClipboard(id, event)
+function loadBingoProgress(sheetToLoad)
+{
+	/* 
+	 * Expected input for sheetToLoad is [1,3,2,5,..]
+	 * where index of array is numeric value of square id
+	 * and cell value is index of ALL_COLOURS 
+	 */
+	if (typeof sheetToLoad !== 'undefined' && sheetToLoad.length) {
+		forEachSquare((i, square) => {
+			$.each(ALL_COLOURS, function(index, colorClass){
+	       square.removeClass(colorClass);
+	    });
+			square.addClass(ALL_COLOURS[sheetToLoad[i]]);
+		});
+	}
+}
+
+var sheetProgressDebounce = null;
+function updateCurrentSheetProgress()
+{
+	if(sheetProgressDebounce) clearTimeout(sheetProgressDebounce);
+	sheetProgressDebounce = setTimeout(function() {
+			updateCurrentSheetProgressDebounced()
+	}, 100);
+}
+
+function updateCurrentSheetProgressDebounced()
+{
+	if (! isLocalStorageIsAvailable()) {
+		return;
+	}
+
+	processSheetAndStoreAs(SHEET_PROGRESS_KEY + generateNewUrlParam().replace(/-\d-\d/, ''));
+}
+
+function processSheetAndStoreAs(localStorageKey)
+{
+	var squares = generateSquareProgress();
+	result = {
+		squares: squares,
+		timestamp: Date.now(),
+		haveMarkedSquares: squares.reduce(function(a,b){return !!a | !!b})
+	};
+	
+	// save only if current sheet have marked squares or already in storage
+	if (result.haveMarkedSquares || localStorage.getItem(localStorageKey)) {
```

Comment:
If progress is saved and loaded only `.haveMarkedSquares` is `true`, then having it as part of `result` isn't useful. It should be just a variable in this function. And `function checkRecentCurrentSheet()` doesn't need to check a value that is always `true`.

Reply:
I've removed usage of `haveMarkedSquares` as result field, but I feel that it could lead to some edge cases that I'm unable to check atm.